app/v2/controllers/V2DeparturesController.scala

```diff
@@ -112,12 +120,20 @@ class V2DeparturesControllerImpl @Inject() (
         implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequest(request)
         withTemporaryFile {
           (temporaryFile, source) =>
-            validationService
-              .validateXML(MessageType.DepartureDeclaration, source)
-              .leftMap(translateValidationError)
+            for {
+              _ <- validationService
+                .validateXML(MessageType.DepartureDeclaration, source)
+                .leftMap(translateValidationError)
+
+              fileSource = FileIO.fromPath(temporaryFile)
+
+              declarationResult <- departuresPersistenceService
+                .saveDeclaration(request.eoriNumber, fileSource)
+                .leftMap(translatePersistenceError)
+            } yield declarationResult
         }.fold[Result](
           baseError => Status(baseError.code.statusCode)(Json.toJson(baseError)),
-          _ => Accepted
+          result => Accepted(Json.obj("movementId" -> result.movementId)) // TODO: do we want the message ID too? Create case object if necessary
```

Comment:
What do we want return to the traders? A proper case class will be created for this, just need to decide whether we want to just return the movement id, or include the message id too. I'm leaning towards both.

Reply:
Created a Hateoas response which contains the message and movement ID